src/api.v2/model/userAccess.js

```diff
@@ -10,10 +13,27 @@ const selectableProps = [
 ];
 
 const basicModelFunctions = generateBasicModelFunctions({
-  tableName,
+  tableName: userAccessTable,
   selectableProps,
 });
 
+const canAccessExperiment = async (userId, experimentId, url, method) => {
+  const results = await sqlClient.get()
+    .select()
+    .from(userAccessTable)
+    .where({ experiment_id: experimentId, user_id: userId });
+
+  // If there is no entry for this user and role, then user definitely doesn't have access
+  if (results.length === 0) {
+    return false;
+  }
+
+  const { access_role: role } = results[0];
```

Comment:
you can  use this `first` when trying to get a single value instead of `[0]` https://github.com/knex/knex/issues/1355#issuecomment-211419629